Source/Core/DolphinTool/ExtractCommand.cpp

```diff
@@ -0,0 +1,313 @@
+
+  const bool mute = options.is_set_by_user("mute");
+  const bool gameonly = options.is_set_by_user("gameonly");
+
+  if (!options.is_set("input"))
+  {
+    fmt::println(std::cerr, "Error: No input image set");
+    return EXIT_FAILURE;
+  }
+  const std::string& input_file_path = options["input"];
+
+  const std::string& output_folder_path = options["output"];
+
+  if (!options.is_set("output") && !options.is_set("list"))
+  {
+    fmt::println(std::cerr, "Error: No output folder set");
+    return EXIT_FAILURE;
+  }
+
+  const std::string& single_file_path = options["single"];
+  const std::string& list_path = options["list"];
+  std::string& specific_partition = const_cast<std::string&>(options["partition"]);
+
+  if (options.is_set("output"))
+    std::filesystem::create_directories(output_folder_path);
+
+  if (gameonly)
+    specific_partition = std::string("data");
+
+  if (const std::unique_ptr<DiscIO::BlobReader> blob_reader =
+          DiscIO::CreateBlobReader(input_file_path);
+      !blob_reader)
+  {
+    fmt::println(std::cerr, "Error: Unable to open disc image");
+    return EXIT_FAILURE;
+  }
+
+  const std::unique_ptr<DiscIO::Volume> disc_volume = DiscIO::CreateVolume(input_file_path);
+
+  if (!disc_volume)
+  {
+    fmt::println(std::cerr, "Error: Unable to open volume");
+    return EXIT_FAILURE;
+  }
+
+  if (!VolumeSupported(*disc_volume))
+    return EXIT_FAILURE;
+
+  if (options.is_set_by_user("list"))
+  {
+    ListVolume(*disc_volume, list_path, specific_partition);
+    return EXIT_SUCCESS;
+  }
+
+  bool extracted_one = false;
+
+  if (disc_volume->GetPartitions().empty())
+  {
+    if (options.is_set("partition"))
+      fmt::println(
+          std::cerr,
+          "Warning: --partition has a value even though this image doesn't have any partitions.");
+
+    if (!options.is_set("single"))
+    {
+      ExtractPartition(*disc_volume, DiscIO::PARTITION_NONE, output_folder_path, mute);
+      extracted_one = true;
+    }
+    else if (auto file_info = GetFileInfo(*disc_volume, DiscIO::PARTITION_NONE, single_file_path);
+             file_info != nullptr)
+    {
+      std::string file;
+      file.append(output_folder_path).append("/");
+
+      file.append(single_file_path);
+
+      if (file_info->IsDirectory())
+        ExtractDirectory(*disc_volume, DiscIO::PARTITION_NONE, single_file_path, output_folder_path,
+                         mute);
+      else
+        ExtractFile(*disc_volume, DiscIO::PARTITION_NONE, single_file_path, file);
+
+      extracted_one = true;
+    }
+  }
+  else
+  {
+    for (DiscIO::Partition& p : disc_volume->GetPartitions())
+    {
+      if (const std::optional<u32> partition_type = disc_volume->GetPartitionType(p))
+      {
+        const std::string partition_name = DiscIO::NameForPartitionType(*partition_type, true);
+
+        if (!specific_partition.empty() &&
+            !Common::CaseInsensitiveEquals(specific_partition, partition_name))
+          continue;
```

Comment:
This is arguable, but when specifying the partition with `-p` I would expect the output path to not actually contain the partition name.

That is, with the command line `dolphin-tool extract -i brawl.rvz -o tmp -p UPDATE`, it generates `tmp/UPDATE/files/__update.inf`, when I would expect the `UPDATE` to be omitted since I have already specified it in the command line.

Reply:
I believe this really comes down to personal preference. I personally prefer outputting the partition name to the final path. Still, for some people using the command purely to extract a couple of files and nothing else, it could become a bit of a nuisance. I would rather have it do it anyway; it would be quite useful for automating stuff, and it doesn't really hurt anyone.